### PR TITLE
Allow to use mouse wheel to change system options

### PR DESCRIPTION
**Pull request by a1exsh**
Mouse wheel is enabled on the following options:

- Music
- Effects
- Hero Speed
- AI Speed
- Scroll Speed

Any adjustments made using the wheel do not wrap over max. value, as opposed
to clicking.  Clicking still works as before.

Also unify the value wrapping math.

### CODE REVIEW

File: src/fheroes2/dialog/dialog_system_options.cpp [hunk 1]
```diff
@@ -266,56 +265,102 @@ namespace
...
+            bool saveSoundVolume = false;
+            if ( Audio::isValid() ) {
+                if ( le.MouseClickLeft( musicVolumeRoi ) ) {
+                    conf.SetMusicVolume( ( conf.MusicVolume() + 1 ) % 10 );
+                    saveMusicVolume = true;
+                }
+                else if ( le.MouseWheelUp( musicVolumeRoi ) ) {
+                    conf.SetMusicVolume( std::min( conf.MusicVolume() + 1, 9 ) );
+                    saveMusicVolume = true;
+                }
```
Comment (ihhub) [inline src/fheroes2/dialog/dialog_system_options.cpp:279]: Shouldn't we combine these two conditions (through logical OR) as they're doing the same thing?

I think the same question for the rest of options.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @a1exsh , I left one comment here. Could you please take a look?

File: src/fheroes2/dialog/dialog_system_options.cpp (hunk 1 above)
Comment (a1exsh) [reply src/fheroes2/dialog/dialog_system_options.cpp:279]: Not exactly: clicking wraps over max value, while scrolling doesn't.

See the linked discussion, maybe we change the behavior as the outcome, then we could simplify the code here. But I do like the implemented behavior better without wrapping on scroll. ;)

Comment (ihhub) [reply src/fheroes2/dialog/dialog_system_options.cpp:279]: Got it. Thanks for the explanation!

**Comment by ihhub** [comment]
@a1exsh , thanks a lot for this improvement!

**Pull request merged**